src/RingGrids/full_grids.jl

```diff
@@ -19,12 +19,25 @@ matrix_size(Grid::Type{<:AbstractFullGridArray}, nlat_half::Integer) =
 
 ## CONVERSION
 # convert an AbstractMatrix to the full grids, and vice versa
-(Grid::Type{<:AbstractFullGrid})(M::AbstractMatrix) = Grid(vec(M))
+"""
+($TYPEDSIGNATURES)
+Initialize an instance of the grid from an Array. For keyword argument `input_as=Vector` (default)
+the leading dimension is interpreted as a flat vector of all horizontal entries in one layer.
+For `input_as==Matrx` the first two leading dimensions are interpreted as longitute and latitude.
+This is only possible for full grids that are a subtype of `AbstractFullGridArray`.
+"""
+(Grid::Type{<:AbstractFullGridArray})(M::AbstractArray; input_as=Vector) = Grid(M, input_as)
+
+function (Grid::Type{<:AbstractFullGridArray})(M::AbstractArray, input_as::Type{Matrix})
+    # flatten the two horizontal dimensions into one, identical to vec(M) for M <: AbstractMatrix
+    M_flat = reshape(M, :, size(M)[3:end]...)
+    Grid(M_flat)
+end 
```

Comment:
Oh also shortened this, as the two additional lines you wrote is exactly what happens if you wrap an array into a grid: get the number of horizontal points `get_npoints2D` then determine what  resolution parameter `nlat_half` that is